R/dice.R

```diff
@@ -199,7 +199,8 @@ setGeneric("dice", function(x,
   names(s)[1] <- hzidn
   
   # FULL JOIN via fast data.table compiled code
-  res <- merge(h, s, by = hzidn, all = TRUE, sort = FALSE)
+  hm <- merge(h, hzMetadata(x), by = c(idn, hzidn), all.x = TRUE, sort = FALSE)
```

Comment:
This isn't going to work in general: the metadata need to be included in `vars` further up, otherwise there will be duplication. This happens when e.g. including the horizon designation in the RHS of the formula or specifying all hz variables with `.`.

```r
set.seed(1010)
d <- lapply(as.character(1:10), random_profile, n = c(6, 7, 8), n_prop = 5, method = 'LPP', SPC = FALSE)
d <- do.call('rbind', d)
depths(d) <- id ~ top + bottom

hzdesgnname(d) <- 'name'

# duplication
dice(d, fm = 25 ~ p1 + name, SPC = FALSE)

# OK
dice(d, fm = 25 ~ p1, SPC = FALSE)

# duplication
dice(d, fm = 25 ~ ., SPC = FALSE)
```

Reply:
You are right, I think this is fixed now with https://github.com/ncss-tech/aqp/commit/721ef8a435e020d28d948583901a72f7cd600ca0